test/check-machines-migrate

```diff
@@ -0,0 +1,226 @@
+        machine2.execute("echo 'migration_host = \"10.111.113.1\"' >> /etc/libvirt/qemu.conf")
+        machine2.execute("systemctl restart libvirtd.service")
+
+        # setup key authentization
+        machine1.execute("mkdir -p $HOME/.ssh")
+        machine1.execute("chmod 0700 $HOME/.ssh")
+        machine1.execute("ssh-keygen -t rsa -N \"\" -f $HOME/.ssh/id_rsa")
+        machine1.execute("echo 'StrictHostKeyChecking no' > $HOME/.ssh/config")
+        machine1PubKey = machine1.execute("cat $HOME/.ssh/id_rsa.pub")
+        machine2.execute("echo '{0}' >> /home/admin/.ssh/authorized_keys".format(machine1PubKey))
+        machine2.execute("mkdir -p $HOME/.ssh")
+        machine2.execute("chmod 0700 $HOME/.ssh")
+        machine2.execute("ssh-keygen -t rsa -N \"\" -f $HOME/.ssh/id_rsa")
+        machine2.execute("echo 'StrictHostKeyChecking no' > $HOME/.ssh/config")
+        machine2PubKey = machine2.execute("cat $HOME/.ssh/id_rsa.pub")
+        machine1.execute("echo '{0}' >> /home/admin/.ssh/authorized_keys".format(machine2PubKey))
+        machine2.execute("usermod --append --groups libvirt admin")
+        machine1.execute("usermod --append --groups libvirt admin")
+
+        xml = POOL_XML.format(path="/var/lib/libvirt/images")
+        machine2.execute("echo \"{0}\" > /tmp/xml && virsh pool-define /tmp/xml && virsh pool-start images".format(xml))
+
+        machine1.execute("systemctl stop firewalld")
+        machine2.execute("systemctl stop firewalld")
+
+    def prepareSharedStorage(self, machine1, machine2):
+        # Set up shared NFS storage available on machine1
+        self.restore_file("/etc/exports")
+        nfs_pool = os.path.join(self.vm_tmpdir, "nfs_pool")
+        mnt_exports = os.path.join(self.vm_tmpdir, "mnt_exports")
+        machine1.execute("mkdir {0} {1} {2} && echo '{2} 10.111.113.2/24(rw,sync,no_root_squash,no_subtree_check,fsid=0)' > /etc/exports".format(self.vm_tmpdir, nfs_pool, mnt_exports))
+        machine1.execute("systemctl restart nfs-server")
+        machine1.execute("virsh pool-define-as nfs-pool --type netfs --target {0} --source-host 10.111.113.2 --source-path {1} && virsh pool-start nfs-pool".format(nfs_pool, mnt_exports))
+        machine1.execute("virsh vol-create-as --pool nfs-pool --name nfs-volume-1 --capacity 1M --format qcow2")
+        machine1.execute("virsh detach-disk --domain subVmTest1 --target vda --live")
+        disk_xml = '''
+<disk type='volume' device='disk'>
+    <driver name='qemu' type='qcow2'/>
+    <source pool='nfs-pool' volume='nfs-volume-1'/>
+    <target dev='vdb' bus='virtio'/>
+</disk>'''
+        machine1.execute("echo \"{0}\" > /tmp/net_disk_xml".format(disk_xml))
+        machine1.execute("virsh attach-device --domain subVmTest1 --file /tmp/net_disk_xml --live")
+
+        machine2.execute("mkdir {0} {1} {2} && echo '{2} 10.111.113.2/24(rw,sync,no_root_squash,no_subtree_check,fsid=0)' > /etc/exports".format(self.vm_tmpdir, nfs_pool, mnt_exports))
```

Comment:
That looks wrong. machine-1 should be the NFS server, machine2 the NFS client no? Why you need the export here as well?

Reply:
Let me deal with that